tools/deviceGen/output/traits_defaults.hlsl

```diff
@@ -0,0 +1,382 @@
+NBL_GENERATE_GET_OR_DEFAULT(maxTessellationEvaluationOutputComponents, uint16_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxGeometryShaderInvocations, uint16_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxGeometryInputComponents, uint16_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxGeometryOutputComponents, uint16_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxGeometryOutputVertices, uint16_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxGeometryTotalOutputComponents, uint16_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxFragmentInputComponents, uint32_t, 116);
+NBL_GENERATE_GET_OR_DEFAULT(maxFragmentOutputAttachments, uint32_t, 8);
+NBL_GENERATE_GET_OR_DEFAULT(maxFragmentDualSrcAttachments, uint32_t, 1);
+NBL_GENERATE_GET_OR_DEFAULT(maxFragmentCombinedOutputResources, uint32_t, 16);
+NBL_GENERATE_GET_OR_DEFAULT(maxComputeSharedMemorySize, uint32_t, 32768);
+NBL_GENERATE_GET_OR_DEFAULT(maxComputeWorkGroupCount[3], uint32_t, {MinMaxWorkgroupCount,MinMaxWorkgroupCount,MinMaxWorkgroupCount});
+NBL_GENERATE_GET_OR_DEFAULT(maxComputeWorkGroupInvocations, uint16_t, MinMaxWorkgroupInvocations);
+NBL_GENERATE_GET_OR_DEFAULT(maxWorkgroupSize[3], uint16_t, {MinMaxWorkgroupInvocations,MinMaxWorkgroupInvocations,64u});
+NBL_GENERATE_GET_OR_DEFAULT(subPixelPrecisionBits, uint8_t, 4);
+NBL_GENERATE_GET_OR_DEFAULT(subTexelPrecisionBits, uint8_t, 4);
+NBL_GENERATE_GET_OR_DEFAULT(mipmapPrecisionBits, uint8_t, 4);
+NBL_GENERATE_GET_OR_DEFAULT(maxDrawIndirectCount, uint32_t, 1073741824);
+NBL_GENERATE_GET_OR_DEFAULT(maxSamplerLodBias, float, 4);
+NBL_GENERATE_GET_OR_DEFAULT(maxSamplerAnisotropyLog2, uint8_t, 4);
+NBL_GENERATE_GET_OR_DEFAULT(maxViewports, uint8_t, 16);
+NBL_GENERATE_GET_OR_DEFAULT(maxViewportDims[2], uint16_t, {MinMaxImageDimension2D,MinMaxImageDimension2D});
+NBL_GENERATE_GET_OR_DEFAULT(viewportBoundsRange[2], float, { -MinMaxImageDimension2D*2u, MinMaxImageDimension2D*2u-1 });
+NBL_GENERATE_GET_OR_DEFAULT(viewportSubPixelBits, uint32_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(minMemoryMapAlignment, uint16_t, 64);
+NBL_GENERATE_GET_OR_DEFAULT(bufferViewAlignment, uint16_t, 64);
+NBL_GENERATE_GET_OR_DEFAULT(minUBOAlignment, uint16_t, 256);
+NBL_GENERATE_GET_OR_DEFAULT(minSSBOAlignment, uint16_t, 64);
+NBL_GENERATE_GET_OR_DEFAULT(minTexelOffset, int8_t, -8);
+NBL_GENERATE_GET_OR_DEFAULT(maxTexelOffset, uint8_t, 7);
+NBL_GENERATE_GET_OR_DEFAULT(minTexelGatherOffset, int8_t, -8);
+NBL_GENERATE_GET_OR_DEFAULT(maxTexelGatherOffset, uint8_t, 7);
+NBL_GENERATE_GET_OR_DEFAULT(minInterpolationOffset, float, -0.5);
+NBL_GENERATE_GET_OR_DEFAULT(maxInterpolationOffset, float, 0.437500000);
+NBL_GENERATE_GET_OR_DEFAULT(subPixelInterpolationOffsetBits, uint8_t, MinSubPixelInterpolationOffsetBits);
+NBL_GENERATE_GET_OR_DEFAULT(maxFramebufferWidth, uint32_t, MinMaxImageDimension2D);
+NBL_GENERATE_GET_OR_DEFAULT(maxFramebufferHeight, uint32_t, MinMaxImageDimension2D);
+NBL_GENERATE_GET_OR_DEFAULT(maxFramebufferLayers, uint32_t, 1024);
+NBL_GENERATE_GET_OR_DEFAULT(maxColorAttachments, uint8_t, MinMaxColorAttachments);
```

Comment:
in HLSL you don't have access to the `constexprs` used in C++ structs.

so substitute the values here for HLSL